torch/_inductor/codegen/wrapper.py

```diff
@@ -353,6 +356,63 @@ def call(args):
                 )
             V.graph.sizevars.codegen(self.wrapper_call, V.graph.graph_inputs)
 
+    def write_mkldnn_param_gen_utils(self):
+        self.header.splice(
+            """
+            class mkldnn_param_cache:
+                def __init__(self):
+                    self.p_cache = dict()
+
+                def add_param(self, param_key, param, param_type):
+                    self.p_cache[param_key] = (param, param_type)
+
+                def get_param(self, param_key):
+                    if param_key in self.p_cache:
+                        return self.p_cache[param_key][0]
+                    else:
+                        return 0
+
+                def check_param(self, param_key):
+                    if param_key in self.p_cache:
+                        return True
+                    else:
+                        return False
+
+                def clear(self):
+                    for param in self.p_cache.values():
+                        torch.mkldnn_delete_param(param[0], param[1])
+                    self.p_cache.clear()
+
+                def __del__(self):
+                    self.clear()
+
+            def gen_mkldnn_cache_key(op_id, *param_args):
+                # Add op id and multi-threading number as part of key
+                key = [op_id, torch.get_num_threads()]
+                for arg in param_args[0]:
+                    # Add inputs tensors' shape as part of key, assume weight/bias shape will not change
```

Comment:
Do we also have unit tests to cover the dynamic shape scenarios?

Reply:
The problem is that seems TorchDynamo seems with problem under dynamic shape=True even with a simple model with only add/sub. Any suggested way to test this?

Reply:
One thing I realized was that we don't have to worry about dynamic shape handling with weight prepacked conv/linear since the weight prepacking needs the concrete input shapes to work. Also, for here, we don't have to check the input shapes and add them input the cache key if the input shapes are static, right?